packages/web-frontend/src/historyNavigation/historyMiddleware.js

```diff
@@ -33,16 +37,25 @@ import {
   attemptPushHistory,
   getInitialLocationComponents,
 } from './historyNavigation';
-import { URL_COMPONENTS } from './constants';
+import { URL_COMPONENTS, PASSWORD_RESET_PREFIX, VERIFY_EMAIL_PREFIX } from './constants';
 
 export const reactToInitialState = store => {
   const { dispatch: rawDispatch } = store;
   const dispatch = action => rawDispatch({ ...action, meta: { preventHistoryUpdate: true } });
 
   const { userPage, projectSelector, ...otherComponents } = getInitialLocationComponents();
   if (userPage) {
-    // TODO: Implemented in userPage PR
-    dispatch(goHome());
+    switch (userPage) {
+      case PASSWORD_RESET_PREFIX:
+        dispatch(setPasswordResetToken(otherComponents[URL_COMPONENTS.PASSWORD_RESET_TOKEN]));
+        dispatch(openUserPage(DIALOG_PAGE_RESET_PASSWORD));
+        break;
+      case VERIFY_EMAIL_PREFIX:
+        dispatch(setVerifyEmailToken(otherComponents[URL_COMPONENTS.VERIFY_EMAIL_TOKEN]));
+        break;
+      default:
+    }
+    return;
```

Comment:
Let's extract this to a new function... Also let's throw an error in the `default` case

Reply:
a) I'm happy to extract this if you much prefer, but I like that all of the 'on load' code is in the same place? Is that a valid reason to have an otherwise-too-long function?

b) Sure, sounds good and makes sense. It will only happen if someone adds something to USER_PAGE_PREFIXES, but doesn't add a handler here, so it should error I suppose.

Reply:
I personally find most functions longer than 15 lines and/or nested more than 3 blocks worth refactoring. If I'm not mistaken, popular bibliography on clean code practices also suggests limits of that sort. After initial code development, I do a code "massage" trying to make code not only readable and accurate, but also "pretty". It may be that a refactoring actually makes things more worse, in which case I'll revert the function to its previous form.

There is no compelling reason to do this here, so I'll leave the final decision to you. In my opinion it would be worth trying both locally and try to imagine which version of the code your future (eg 6 months from now) self would prefer to read.

About b), yes let's do this for the exact reason that you mentioned

Reply:
I've had a look and am still not sure, but in the absence of a strong opinion from me I will defer to your expertise (forgot to comment this earlier).